src/crimson/os/seastore/seastore.h

```diff
@@ -210,6 +212,7 @@ class SeaStore final : public FuturizedStore {
   TransactionManagerRef transaction_manager;
   CollectionManagerRef collection_manager;
   OnodeManagerRef onode_manager;
+  PerfService &perf_service;
```

Comment:
SeaStore needs to actually own this.  Pass it into the constructor by value and this should be

```
PerfService perf_service;
```

Reply:
I'd also be fine with constructing it in place here and not passing it into the constructor.

Reply:
@athanatos seastore constructor need TransactionManagerRef and TransactionManagerRef constructor need perfcounter reference.  If I move perf_service initialized in seastore, then TransactionManager should be also initialized in seastore.  and TransactionManager  need journal cache initialized first.   So which do you prefer? initialize all TransactionManager  and PerfService outside of seastore constructor or move them into seastore ?

Reply:
Ah.  In that case you'll have to pass it in by value/rval reference.  SeaStore still needs to own it just as it does transaction_manager, collection_manager, onode_manager etc.